### ISSUE TITLE
Support zookeeper rolling upgrades

### ISSUE THREAD

**Issue by spiegela** [pravega/zookeeper-operator#5]
Currently rolling updates _should_ work, but have not been tested. Rolling updates should be tested an validated.

**Comment by fpj** [comment]
@spiegela could you elaborate on what kind of rolling upgrade support you are referring to when you say that it should work?

**Comment by EronWright** [comment]
I assume that rolling upgrade here means to upgrade the ZK servers in an online fashion without downtime.  This is said to be possible with ZK 3.5+.   Last time I looked at the documentation, I got the impression that the procedure must, at each step, wait for the ZK server to rejoin the ensemble (specifically by looking at the results of `stat` command).   To interoperate with `StatefulSet` and its rolling upgrade feature, we must tie the pod readiness status to that.

**Comment by pbelgundi** [comment]
I tried upgrading a zookeeper cluster from its earlier version to a newer version ( from 0.2.1 to 0.2.2 ) using `kubectl edit zk <zookeeper-clusr-name>` and the upgrade completed successfully without any errors.
Post upgrade, the zk image version displayed on pods and the zk object was 0.2.2 and leader election completed successfully post upgrade and all zk servers were running successfully as participants.

**Comment by langruixiang** [comment]
I tried rolling upgrade the cluster by `kubectl edit` pullPolicy field (just in order to trigger a rolling update). First time, it succeed as @pbelgundi said. Second time, the `pod/example-1` can't get ready for a long time (even more than 2h). 
<img width="680" alt="zk_rollout_pod" src="https://user-images.githubusercontent.com/15683282/65758592-e9a1c280-e14b-11e9-93c4-fd0770121ebd.png">
Here is the log of `pod/example-1` :
[zk_operator_pod_1.log](https://github.com/pravega/zookeeper-operator/files/3661847/zk_operator_pod_1.log)
After I removing `pod/example-1`, the rolling upgrade continued and succeed.

I terminate the operator and tried `kubectl rollout restart sts example`, I got the same result. ( If 
 don't terminate the operator, the rollout process will stop after `pod/example-2` got ready. I guess operator stop the rollout process. )
